File: intervals/cppRealizations/meetingRooms.cpp
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;


class Interval {
public:
    int start, end;
    Interval(int start, int end) {
        this->start = start;
        this->end = end;
    }
};


class Solution {
public:
    bool canAttendMeetings(vector<Interval>& intervals) {
        if(intervals.size() <= 1) return true;
        sort(intervals.begin(), intervals.end(), [](const Interval& a, const Interval& b) {
            return a.start < b.start;
        });
        for(int i = 0; i < intervals.size() - 1; i++){
            if(intervals[i].end > intervals[i + 1].start){
                return false;
            }
        }
        return true;
    }
};



int main(){
    vector<Interval> intervals = {Interval(0, 30), Interval(5, 10), Interval(15, 20)};
    // vector<Interval> intervals = {{0, 30}, {5, 10}, {15, 20}};
    cout << "Output:" << endl;
    cout << Solution().canAttendMeetings(intervals) << endl;
    return 0;
}

File: intervals/meetingRooms.py
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end


class Solution:
    def canAttendMeetings(self, intervals: list[Interval]) -> bool:
        intervals.sort(key=lambda x: x.start)

        for i in range(1, len(intervals)):
            interval1 = intervals[i - 1]
            interval2 = intervals[i]
            if interval1.end > interval2.start:
                return False

        return True


if __name__ == "__main__":
    intervals = [Interval(0, 30), Interval(5, 10), Interval(15, 20)]
    print("Solution:")
    print(Solution().canAttendMeetings(intervals))